Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2685 -- Move subject page search box to filter region

Resolves [EREGCSC-2685](https://jiraent.cms.gov/browse/EREGCSC-2685)

**Description**

Removes ability to search by query string on the Subjects page.  The Search Input still exists on the page, but it is moved out of the left sidebar and to the top center of the Subjects page.  When a user searches by query string, they are redirected away from the Subjects page to the Search page with only their selected subject preserved.

**This pull request changes:**

- Moves Search Input element from left sidebar of Subjects page to top center of page with the other filters (Choose Category, Document Type selector)
- Removes `q` param from URL if user attempts to visit Subjects page with a link that includes `q` param
- Removes/refactors all logic on the Subjects page associated with search querystrings
- Updates Subjects end to end test suite to remove assertions that tested searching by query string
- Adds new end to end tests and assertions that verify the redirection to the Search page
- Various narrow/mobile style tweaks
- Includes `fetchCounts` fix: the counts on the Subject page are now based on the selected subject, while the counts on the Search page are still based on the search query

**Steps to manually verify this change:**

1. Green check marks
2. [Visit experimental deployment](https://bm3gws9jr4.execute-api.us-east-1.amazonaws.com/dev1461)
3. Visit `/subjects` page and select a subject
   - Notice that the search input is no longer in the left sidebar; it is at the top center of the screen under the other filters (Category select, Document Type checkboxes)
4. The placeholder test in the search input should read "Search within X" where X is the name of the selected subject
5. Select a category
6. Type a search query into the Search Input and click submit/press enter
7. You should be taken to the Search page for that search query
   - The subject should be preserved
   - The category and document type should NOT be preserved.  No categories or document types should be selected
